packages/neo-one-client-core/src/user/DapiUserAccountProvider.ts

```diff
@@ -0,0 +1,577 @@
+            label,
+          });
+        })
+        .catch(() => {
+          // do nothing
+        });
+    });
+    this.dapi.addEventListener('NETWORK_CHANGED', (dapiNetworks) => {
+      this.updateNetworks(dapiNetworks);
+    });
+    this.dapi.addEventListener('DISCONNECTED', () => {
+      throw new Error(`The account connected to the dapp via the dapi provider has been disconnected (logged out).`);
+    });
+
+    this.userAccounts$ = this.userAccountsInternal$;
+    this.currentUserAccount$ = this.currentAccountInternal$.pipe(distinctUntilChanged());
+    this.networks$ = this.networksInternal$;
+  }
+
+  public getCurrentUserAccount(): UserAccount | undefined {
+    if (!this.mutableInitialized) {
+      throw new NotIntializedError();
+    }
+
+    return this.currentAccountInternal$.getValue();
+  }
+
+  public async selectUserAccount(_userAccountID?: UserAccountID): Promise<void> {
+    throw new NotImplementedError('selectUserAccount');
+  }
+
+  public async deleteUserAccount(_userAccountID?: UserAccountID): Promise<void> {
+    throw new NotImplementedError('deleteUserAccount');
+  }
+
+  public getUserAccounts(): ReadonlyArray<UserAccount> {
+    if (!this.mutableInitialized) {
+      throw new NotIntializedError();
+    }
+
+    return this.userAccountsInternal$.getValue();
+  }
+
+  public getNetworks(): ReadonlyArray<NetworkType> {
+    if (!this.mutableInitialized) {
+      throw new NotIntializedError();
+    }
+
+    return this.networksInternal$.getValue();
+  }
+
+  public async updateUserAccountName(_options: UpdateAccountNameOptions): Promise<void> {
+    throw new NotImplementedError('updateUserAccountName');
+  }
+
+  public async getBlockCount(network: NetworkType): Promise<number> {
+    await this.initPromise;
+
+    return this.provider.getBlockCount(network);
+  }
+
+  public async getAccount(network: NetworkType, address: AddressString): Promise<Account> {
+    await this.initPromise;
+    const balances = await this.dapi.getBalance({ params: { address, assets: [Hash256.NEO, Hash256.GAS] }, network });
+
+    return {
+      address,
+      balances: balances[address].reduce<{ readonly [asset: string]: BigNumber }>(
+        (acc, balance) => ({
+          ...acc,
+          [balance.assetID]: new BigNumber(balance.amount),
+        }),
+        {},
+      ),
+    };
+  }
+
+  public iterBlocks(network: NetworkType, options?: IterOptions): AsyncIterable<Block> {
+    if (!this.mutableInitialized) {
+      throw new NotIntializedError();
+    }
+
+    return this.provider.iterBlocks(network, options);
+  }
+
+  public iterActionsRaw(network: NetworkType, options?: IterOptions): AsyncIterable<RawAction> {
+    if (!this.mutableInitialized) {
+      throw new NotIntializedError();
+    }
+
+    if (this.provider.iterActionsRaw === undefined) {
+      throw new NotImplementedError('iterActionsRaw');
+    }
+
+    return this.provider.iterActionsRaw(network, options);
+  }
+
+  public async transfer(
```

Comment:
Same comment here, try to move this into the base class and call protected abstract methods. 

Also, we can probably support multiple transfers by just calling the dapi multiple times in sequence.